### ISSUE TITLE
RealmJS initialization clears all caches making cocoa realm object invalid

### ISSUE THREAD

**Issue by Tmac90301** [realm/realm-js#3487]
## Bugs: 
In our hybrid project react native part uses RealmJS as database SDK and now we also want use realm cocoa in our native part. We find that in RealmJS initialization will clear all caches and this makes our native realm object all becoming **invalid**. Once we try to access these invalid object again, it causes a crash noted "Object has been deleted or invalidated.". 

## Expected Results

RealmJS works compatiably with Realm Cocoa in one project.

## Actual Results

Open react native module using RealmJS clears all realm cache and makes native realm object not available which results in crash.

```
Terminating app due to uncaught exception 'RLMException', reason: 'Object has been deleted or invalidated.'
```

## Code Sample

```c
void _initializeOnJSThread(JSContextRefExtractor jsContextExtractor) {
    // Make sure the previous JS thread is completely finished before continuing.
    static __weak NSThread *s_currentJSThread;
    while (s_currentJSThread && !s_currentJSThread.finished) {
        [NSThread sleepForTimeInterval:0.1];
    }
    s_currentJSThread = [NSThread currentThread];

    // Close all cached Realms from the previous JS thread.
    realm::_impl::RealmCoordinator::clear_all_caches();

    RJSInitializeInContext(jsContextExtractor());
}
```

## Version of Realm and Tooling

- Realm JS SDK Version: 6.1.2
- Realm Cocoa SDK Version: 5.4.1
- Node or React Native: node v14.12.0, react native cli 2.0.1
- Client OS & Version: macOS 10.15.6

**Comment by kneth** [comment]
Thank you for reporting. True, Realm JavaScript enables caching (and clears the cache). In recent versions, we have move the clearing of the cache to when the app shuts down: https://github.com/realm/realm-js/blob/master/react-native/ios/RealmReact/RealmReact.mm#L270-L279

**Comment by Tmac90301** [comment]
Moving the clearing of the cache to when the app shuts down may cause the same problem, because Realm Cocoa uses same caches with RealmJS. Once RealmJS clears the cache, all Realm Cocoa cached objects become invalid or deleted and can no more be used.

**Comment by kneth** [comment]
Mixture two Realm SDKs seem to be non-trivial.

**Comment by bmunkholm** [comment]
I don't believe there is a huge demand for this feature, but we will leave it open for upvotes for a while to gather interest.
